src/components/balances/principal-token-balances.tsx

```diff
@@ -57,14 +71,33 @@ const tabs: { label: string; slug: 'ft' | 'nft' }[] = [
   { label: 'Collectibles', slug: 'nft' },
 ];
 
-export const TokenBalancesCard: React.FC<FlexProps & { balances: AddressBalanceResponse }> = ({
-  balances,
-  ...rest
-}) => {
+export const TokenBalancesCard: React.FC<
+  FlexProps & { balances: AddressBalanceResponse; nftHoldings?: NonFungibleTokenHoldingsList }
+> = ({ balances, nftHoldings, ...rest }) => {
   const [activeTab, setActiveTab] = React.useState<'ft' | 'nft'>('ft');
 
   const TabContent = activeTab === 'ft' ? FtBalances : NftBalances;
 
+  const bnsHexValues = useMemo(
+    () =>
+      nftHoldings?.results
+        ?.filter(nftHolding => nftHolding.asset_identifier.endsWith('.bns::names'))
+        ?.reduce((acc, data) => {
+          acc[data.asset_identifier] = data.value?.hex
+            ? {
+                name: hexToString(
```

Comment:
`hexToString` doesn't handle `undefined` inputs well, and the optional chaining here suggests it's possible for the input to be undefined -- maybe there should be a nice default display value to be on the safe side